llvm/lib/AsmParser/LLLexer.cpp

```diff
@@ -251,6 +258,12 @@ lltok::Kind LLLexer::LexToken() {
     case ',': return lltok::comma;
     case '*': return lltok::star;
     case '|': return lltok::bar;
+    case '/':
+      if (peekNextChar() != '*')
```

Comment:
This is the fix. The old code was:

```
      if (peekNextChar() == '*' && SkipCComment())
        return lltok::Error;
      continue;
```

Given the input:

```
@B = external global i32

*/
```

The `@B = external global i32*` is a valid sequence of tokens (The * becomes a part of the i32 * type). The `/` following the * is silently eaten by the old code due to the `continue`. The fix now correctly executes `continue` only when it has successfully parsed a C style comment.

Reply:
Right, Thanks. I have reverted that part of the change and the code is now using getNextChar() directly. That entailed handling EOF following a * within a comment, so added another unit test for that.